config/configmodels/configmodels.go

```diff
@@ -44,15 +44,50 @@ type Config struct {
 type NamedEntity interface {
 	Name() string
 	SetName(name string)
+	Type() string
+	SetType(typeStr string)
+	IsEnabled() bool
+}
+
+// Entity implements NamedEntity.
+type Entity struct {
```

Comment:
Ideally receivers, extensions, etc. would use the same Entity struct but unfortunately it'd require a breaking API change.